app/helpers/Intersect.js

```diff
@@ -415,3 +451,43 @@ export function disableDebugMode() {
   // Kick off intersection calculations to restore data
   checkIntersect();
 }
+
+/**
+ * Records where user has crossed paths to avoid repeated notifications or checking
+ * @param {object} intersections - The user intersections history
+ * @param {object} currentLocation - The user's location
+ * @param {object} concernLocation - The infected location
+ */
+function addNewIntersection(intersections, currentLocation, infectedLocation) {
+  let { time } = currentLocation;
+
+  if (time in intersections) {
+    intersections[time].push(infectedLocation);
+  } else {
+    intersections[time] = [infectedLocation];
+  }
+}
+
+/**
+ * Check if intersection has already been identified
+ * @param {object} intersections - The user intersections history
+ * @param {object} currentLocation - The user's location
+ * @param {object} concernLocation - The infected location
+ */
+function hasExistingIntersection(
+  intersections,
+  currentLocation,
+  concernLocation,
+) {
+  let { time } = currentLocation;
+
+  return (
+    time in intersections &&
+    intersections[time].some(
```

Comment:
This could become a performance issue if there are lots of prior intersections at the same time value... @kenpugsley can you weigh in here? is that likely?

Reply:
I was thinking of implementing a binary search similar to the one being used to find the minimum time for concern locations. The known intersections will need to be sorted however as the time can be unordered. Let me know what you think. @tstirrat @kenpugsley @tremblerz

Reply:
Code has been updated. Instead of storing a list of intersections at a given time, we could store the infected location time as the key to look up the location data (longitude and latitude). To check if an intersection has been identified, it would be as simple as this:
```
return (
    knownIntersections[currentLocationTime] &&
    knownIntersections[currentLocationTime][concernLocationTime]
  )
```